src/Google.Datastore.V1Beta3/DatastoreDb.cs

```diff
@@ -119,34 +125,49 @@ public class DatastoreDb
         /// using default settings.</param>
         /// <returns>A <see cref="DatastoreDb"/> operating on the specified partition.</returns>
         public static DatastoreDb Create(string projectId, string namespaceId = "", DatastoreClient client = null) =>
-            new DatastoreDb(projectId, namespaceId, client ?? DatastoreClient.Create());
-
-        private DatastoreDb(string projectId, string namespaceId, DatastoreClient client)
-        {
-            _projectId = GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
-            _partitionId = new PartitionId(projectId, GaxPreconditions.CheckNotNull(namespaceId, nameof(namespaceId)));
-            Client = GaxPreconditions.CheckNotNull(client, nameof(client));
-        }
+            new DatastoreDbImpl(projectId, namespaceId, client ?? DatastoreClient.Create());
 
         /// <summary>
         /// Creates a key factory for root entities in this objects's partition.
         /// </summary>
         /// <param name="kind">The kind of entity key to create. Must not be null.</param>
         /// <returns>A key factory with the specified kind and this object's partition.</returns>
-        public KeyFactory CreateKeyFactory(string kind) => new KeyFactory(_partitionId, kind);
+        public virtual KeyFactory CreateKeyFactory(string kind)
```

Comment:
On Wed, Jun 8, 2016 at 12:35 PM, Jon Skeet <notifications@github.com> wrote:

> In src/Google.Datastore.V1Beta3/DatastoreDb.cs
> <https://github.com/GoogleCloudPlatform/gcloud-dotnet/pull/193#discussion_r66323982>
> :
>
> >
> >          /// <summary>
> >          /// Creates a key factory for root entities in this objects's partition.
> >          /// </summary>
> >          /// <param name="kind">The kind of entity key to create. Must not be null.</param>
> >          /// <returns>A key factory with the specified kind and this object's partition.</returns>
> > -        public KeyFactory CreateKeyFactory(string kind) => new KeyFactory(_partitionId, kind);
> > +        public virtual KeyFactory CreateKeyFactory(string kind)
>
> Nope - but even if it happens to work, that would still only help if you
> only ever wanted the "default" partition of the project ID you're using and
> no namespace.
>
I think that will be the most common case.  Especially for beginners.


> (I'm going to ask the Datastore team, but I'd advise anyone writing an app
> to make the partition configurable, which means that the information has to
> be *somewhere*.)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-dotnet/pull/193/files/c6fb88e98632cb71f3d1d59df4e4a1a2ae4fb9e6#r66323982>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALO54k6RPmCELPsJmYORkTqKrQUQic98ks5qJxmLgaJpZM4IvwZ0>
> .
>